### PR TITLE
 docs: document in-memory joins in optimization docs

### LINKED ISSUE #734: Update optimization guide with auto in-memory joins
After this is merged and updated here: https://github.com/src-d/go-mysql-server/pull/638

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #734

### CODE REVIEW

File: docs/using-gitbase/optimize-queries.md [hunk 1]
```diff
@@ -57,6 +58,24 @@ Some performance issues might not be obvious, but there are a few that really st
 
 - Joins not squashed. If you performed some joins between tables and instead of a `SquashedTable` node you see `Join` and `Table` nodes, it means the joins were not successfully squashed. There is a more detailed explanation about this in next sections of this document.
 - Indexes not used. If you can't see the indexes in your table nodes, it means somehow those indexes are not being used by the table. There is a more detailed explanation about this in next sections of this document.
+- Joins not squashed that are not being executed in memory. There is a more detailed explanation about this in the next sections of this document.
+
+## In-memory joins
+
+There are two modes in which gitbase can execute an inner join:
+
+- Multipass: it fully iterates the right side of the join one time for each row in the left side. This, as you can imagine, it's really expensive, but avoids having to load one side fully in memory.
```
Comment (kuba--) [inline docs/using-gitbase/optimize-queries.md]: maybe:
> ...As you can imagine, this is really expensive, but avoids...

File: docs/using-gitbase/optimize-queries.md [hunk 2]
```diff
@@ -57,6 +58,24 @@ Some performance issues might not be obvious, but there are a few that really st
...
+- In-memory: loads the whole right side in memory and iterates the left side. Both sides are iterated exactly once, thus making the query much faster, but it has the disadvantage of potentially requiring a lot of memory.
```
Comment (kuba--) [inline docs/using-gitbase/optimize-queries.md]: maybe 
> ...thus it makes the query much faster, but it has the disadvantage of potentially requiring a lot of memory.

File: docs/using-gitbase/optimize-queries.md [hunk 3]
```diff
@@ -57,6 +58,24 @@ Some performance issues might not be obvious, but there are a few that really st
...
+
+The default mode is multipass, unless the right side fits in memory (there's a more elaborated explanation about this below).
```
Comment (kuba--) [inline docs/using-gitbase/optimize-queries.md]: > elaborate explanation

**Comment by erizocosmico** [comment]
Fixed @kuba--

**Pull request merged**